### PR TITLE
ELF64 support for DetourBackend

### PR DESCRIPTION
**Pull request by jkrshnmenon**

### CODE REVIEW

File: patcherex/backends/detourbackend.py [hunk 1]
```diff
@@ -315,22 +315,43 @@ def setup_headers(self,segments):
         self.ncontent = self.ncontent.ljust(len(self.ncontent)+self.additional_headers_size, "\x00")
 
     def dump_segments(self, tprint=False):
-        # from: https://github.com/CyberGrandChallenge/readcgcef/blob/master/readcgcef-minimal.py
-        header_size = 16 + 2*2 + 4*5 + 2*6
+        header_size = 4 + 1*5 + 7 + 2 + 2 + 4
+        bits = self.project.arch.bits/8
```
Comment (schieb) [inline patcherex/backends/detourbackend.py]: bits should be bytes. But the again, why not just use arch.bytes throughout?

Comment (jkrshnmenon) [reply patcherex/backends/detourbackend.py]: Fair enough. I'll do that.

**Comment by ltfish** [comment]
The code looks good to me. Can you please provide at least one test case? A smoketest is enough so we know we won't accidentally break it (especially during the Python 3 migration).

**Comment by jkrshnmenon** [comment]
> The code looks good to me. Can you please provide at least one test case? A smoketest is enough so we know we won't accidentally break it (especially during the Python 3 migration).

The code works with the CADET_00003 binary and the patching example given in the Readme.
But I tried patching an ELF32 bit binary which successfully inserts the patch, but execution gives me this : 
Inconsistency detected by ld.so: rtld.c: 1191: dl_main: Assertion `GL(dl_rtld_map).l_libname' failed!

**Comment by jkrshnmenon** [comment]
> But I tried patching an ELF32 bit binary which successfully inserts the patch, but execution gives me this :
Binary and source in tests/new_test_files/
Python script is tests/test_newdetourbackend.py

**Comment by ltfish** [comment]
Hi, do you want to re-open this PR or create a new PR?

**Comment by jkrshnmenon** [comment]
I'll try to work with the new version of Patcherex and create a new PR.
In the mean time can you let me know if [this](https://github.com/jkrshnmenon/patcherex/tree/testing) is an approach that I should continue with ?

**Comment by ltfish** [comment]
What approach? Are you referring to the new commit that your repo has over angr/patcherex?

**Comment by jkrshnmenon** [comment]
Oh, sorry for the confusion. Yes, I was referring to that commit.

**Comment by ltfish** [comment]
I like it. I wouldn't import IPython at the very beginning though (since angr does not, and should not rely on it).

**Comment by jkrshnmenon** [comment]
IPython is just temporary.
I'm just using it for debugging purposes. I'll remove it before I create the PR.

**Pull request closed without merging**